nvtabular/workflow.py

```diff
@@ -947,6 +966,10 @@ def ddf_to_dataset(
         if output_format not in ("parquet", "hugectr"):
             raise ValueError("Only parquet/hugectr output supported with Dask.")
         ddf = self.get_ddf()
+        # Special dtype conversion
+        if dtypes:
+            _meta = _set_dtypes(ddf._meta, dtypes)
+            ddf = ddf.map_partitions(_set_dtypes, dtypes, meta=_meta)
```

Comment:
If we are going to do a `map_partitions` call, we might as well move this up into `build_and_process_graph` in case the user is not actually writing out a dataset (but wants to use the workflow for a dataloader, or some other processing).

If we are only planning to use the `dtypes` when we write out a transformed dataset, then we should probably say so in the docstring.  Also, ff this is the case, we may want to pass the `dtypes` argument all the way down to the reader to avoid adding an extra task for every partition in ddf.

Reply:
Moved up to `build_and_process_graph `